### PR TITLE
remove `size_scattering` kwarg from `scattering1d`

### LINKED ISSUE #836: `size_scattering` is not needed in `scattering1d`
as discussed on Discord with @MuawizChaudhary 

All our `Scattering1D` frontends have this piece of code in the forward pass

```
if self.vectorize:
    size_scattering = precompute_size_scattering(
        self.J, self.Q, self.T, max_order=self.max_order, detail=True)
else:
    size_scattering = 0
    
S = scattering1d(x, self.backend.pad, self.backend.unpad, self.backend, self.log2_T, self.psi1_f, self.psi2_f,
         self.phi_f, max_order=self.max_order, average=self.average, pad_left=self.pad_left,
         pad_right=self.pad_right, ind_start=self.ind_start, ind_end=self.ind_end,
         oversampling=self.oversampling,
         vectorize=self.vectorize,
         size_scattering=size_scattering,
         out_type=self.out_type)
```

But in fact we could measure `size_scattering` directly in the `scattering1d` routine, without this call to `precompute_size_scattering`. This should simplify our 1D codebase, by removing kwargs and boilerplate code.

### PR DESCRIPTION
**Pull request by lostanlen**
Fixes #836

One kwarg less from `scattering1d` in base_frontend.py`
Fewer LOCs, fewer repetitions

### CODE REVIEW

File: kymatio/scattering1d/core/scattering1d.py [hunk 1]
```diff
@@ -160,9 +151,7 @@ def scattering1d(x, pad, unpad, backend, log2_T, psi1, psi2, phi, pad_left=0,
                     else:
                         S_2 = unpad(U_2_m, ind_start[k1 + k2], ind_end[k1 + k2])
 
-                    out_S_2.append({'coef': S_2,
-                                    'j': (j1, j2),
-                                    'n': (n1, n2)})
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/core/scattering1d.py:165]: personally I found it nicer having this be on multiple lines than just one line, but thats just me

**Review by MuawizChaudhary — COMMENTED** [review]
Does the Keras tests pass?

File: kymatio/scattering1d/core/scattering1d.py (hunk 1 above)
Comment (lostanlen) [reply kymatio/scattering1d/core/scattering1d.py:165]: should i undo?

Comment (MuawizChaudhary) [reply kymatio/scattering1d/core/scattering1d.py:165]: if you can that would be nice

**Review by MuawizChaudhary — APPROVED** [review]
I approve these changes.

**Comment by lostanlen** [comment]
OK, please merge at your convenience.
(preferably soon since i'd like to rebase #839 and work on removing `vectorize`)

**Comment by MuawizChaudhary** [comment]
This passes the checks and it passes the test on my local system. I'm squashing and merging.

**Pull request merged**